src/controllers/userController.js

```diff
@@ -367,15 +367,9 @@ class userController {
 			body
 		} = req;
 		try {
-			const user = await Models.Users.findOne({ where: { email: managerEmail } });
-
-			if (!user || user.role !== 'manager') {
-				return res.status(404).json({
-					status: 404,
-					error: setLanguage(preferedLang).__('managerEmailNotExist')
-				});
-			}
-
+			const {isUpdated, managerEmail} = await Models.Users.findOne({ where: { id } });
+			const automatedEmail = isUpdated?managerEmail
+			:getManagerRandomly(await Models.Users.findAll({ where: { role: 'manager' } })).email;
```

Comment:
@hakoemmy please elaborate on what this findOne is for.

Reply:
@AbdoulNuru what I am doing, manager assignment must happen on the first attempt of updating profile otherwise it won't change users' manager. This is a kind of extra logic to keep track of the same.

Hope it makes sense, if not please let me know for further clarifications

Reply:
@hakoemmy, if `isUpdated` is false you'll get array of users, is that what you want? You either use `findOne` or elaborate the logic again

Reply:
@hakoemmy Thanks, I got it.

Reply:
@AJAkimana I think he is using `findAll` because he has set the `getManagerRandomly` function to receive a list of managers then the function randomly selects one.